presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -328,6 +331,22 @@ protected void setup(Binder binder)
 
         // dynamic filter summary
         jsonCodecBinder(binder).bindJsonCodec(DynamicFilterSummary.class);
+        binder.bind(DynamicFilterService.class).in(Scopes.SINGLETON);
+        jaxrsBinder(binder).bind(DynamicFilterResource.class);
+
+        if (featuresConfig.isDynamicPartitionPruningEnabled()) {
+            binder.bind(new TypeLiteral<Optional<DynamicFilterClientSupplier>>() {}).toProvider(DynamicFilterClientFactoryProvider.class).in(Scopes.SINGLETON);
+            httpClientBinder(binder).bindHttpClient("dynamicFilter", ForDynamicFilterSummary.class)
+                    .withTracing()
```

Comment:
I'd leave it, at least in the experimental phase of the feature, in case we need to debug  the DPP.